pkg/agent/agent.go

```diff
@@ -30,6 +30,56 @@ import (
 	"github.com/kinvolk/seccompagent/pkg/registry"
 )
 
+func closeStateFds(recvFds []int) {
+	// If performance becomes an issue, we can fallback to the new syscall closerange().
+	for i := range recvFds {
+		// Ignore the return code. There isn't anything better to do.
+		unix.Close(i)
+	}
+}
+
+// parseContainerProcessState returns the seccomp-fd and closes the rest of the fds in recvFds.
+// In case of error, no fd is closed.
+// StateFds is assumed to be formated as specs.ContainerProcessState.Fds and
+// recvFds the corresponding list of received fds in the same SCM_RIGHT message.
+func parseStateFds(stateFds []string, recvFds []int) (uintptr, error) {
+	// Lets find the index in stateFds of the seccomp-fd.
+	idx := -1
+	err := false
+
+	for i, name := range stateFds {
+		if name == specs.SeccompFdName && idx == -1 {
+			idx = i
+			continue
+		}
+
+		// We found the seccompFdName two times. Error out!
+		if name == specs.SeccompFdName && idx != -1 {
+			err = true
+		}
+	}
+
+	if idx == -1 || err {
+		return 0, fmt.Errorf("seccomp fd not found or malformed containerProcessState.Fds")
+	}
```

Comment:
Would this be more concise / readable?

```suggestion
	// Lets find the index in stateFds of the seccomp-fd.
	idx := -1
	idxCount := 0
	
	for i, name := range stateFds {
		if name == specs.SeccompFdName {
			idx = i
			idxCount++
		}
	}
	if idxCount != 1 {
		return 0, errors.New("seccomp fd not found or malformed containerProcessState.Fds")
	}
```

Reply:
Why not.

But it is buggy, checking for idxCount 1= 1 doesn't fail if `name == specs.SeccompFdName` is never true. Making it instead `if idxCount != 1 || idx == -1` :)